Source/AMRInterpolator/SimpleInterpSource.hpp

```diff
@@ -0,0 +1,62 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef SIMPLEINTERPSOURCE_H_
+#define SIMPLEINTERPSOURCE_H_
+
+#include "InterpSource.hpp"
+
+// Simple interpolation source where all grid points are in the same "Box"
+// Relevant to be able to use Interpolation classes by themselves
+template <int N_DIMS> class SimpleInterpSource : public InterpSource<N_DIMS>
+{
+    std::array<int, N_DIMS> m_points_per_dir;
+    std::array<bool, N_DIMS> m_is_periodic;
+    std::array<double, N_DIMS> m_dxs;
+
+    LevelData<FArrayBox> *m_fake;
+
+  public:
+    SimpleInterpSource() {}
+    SimpleInterpSource(std::array<int, N_DIMS> a_points_per_dir,
+                       std::array<double, N_DIMS> a_dxs,
+                       std::array<bool, N_DIMS> a_is_periodic = {false})
+    {
+        set(a_points_per_dir, a_dxs, a_is_periodic);
+    }
+
+    void set(std::array<int, N_DIMS> a_points_per_dir,
+             std::array<double, N_DIMS> a_dxs,
+             std::array<bool, N_DIMS> a_is_periodic = {false})
+    {
+        m_points_per_dir = a_points_per_dir;
+        m_dxs = a_dxs;
+        m_is_periodic = a_is_periodic;
+    }
+
+    const LevelData<FArrayBox> &
+    getLevelData(const VariableType var_type = VariableType::evolution) const
+    {
+        CH_assert("This should not be needed.");
+        return *m_fake;
+    }
```

Comment:
I am not a fan of this. It's quite hacky. Also I'm not sure whether a string evaluates to `true` or `false`! Since the `InterpAlgo` classes only need the `get_dxs()` and `contains()` members of `InterpSource`, maybe it should be refactored into a base (needed by the `InterpAlgo` classes) and child (needed by the `AMRInterpolator` class)? For example
```cpp
class LevelDataInterpSource : public InterpSource
```
Alternatively, I'm not even sure of the point of `AMRInterpolator` having a generic `InterpSource` anyway so we could just make it explicitly use `GRAMRLevel` and remove the `getLevelData()` function from `InterpSource`?